nbgrader/tests/test_nbextension.py

```diff
@@ -0,0 +1,200 @@
+import subprocess as sp
+import tempfile
+import os
+import shutil
+
+from nose.tools import assert_equal
+
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support.ui import Select
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.common.keys import Keys
+
+class TestCreateAssignmentNbExtension(object):
+
+    @classmethod
+    def setup_class(cls):
+        cls.tempdir = tempfile.mkdtemp()
+        cls.ipythondir = tempfile.mkdtemp()
+        cls.origdir = os.getcwd()
+        os.chdir(cls.tempdir)
+
+        cls.nbserver = sp.Popen([
+            "ipython", "notebook",
+            "--ipython-dir", cls.ipythondir,
+            "--no-browser",
+            "--port", "9000"], stdout=sp.PIPE, stderr=sp.STDOUT)
+
+    @classmethod
+    def teardown_class(cls):
+        cls.nbserver.kill()
+
+        os.chdir(cls.origdir)
+        shutil.rmtree(cls.tempdir)
+        shutil.rmtree(cls.ipythondir)
+
+    def setup(self):
+        shutil.copy(os.path.join(os.path.dirname(__file__), "files", "blank.ipynb"), "blank.ipynb")
+        self.browser = webdriver.PhantomJS()
```

Comment:
@takluyver it looks like @jhamrick 's implementation uses Phantom, which from what I understand is part of our problem with the current slimer system (CC @minrk).  @jhamrick have you tried the other drivers they have available?  http://docs.seleniumhq.org/projects/webdriver/  If so, were there any caveats?